Makefile

```diff
@@ -0,0 +1,95 @@
+PKGS := $(shell go list ./... | grep -v /vendor)
+
+BIN_DIR := $(GOPATH)/bin
+
+# Try to detect current branch if not provided from environment
+BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
+
+# Commit hash from git
+COMMIT=$(shell git rev-parse --short HEAD)
+
+# Tag on this commit
+TAG = $(shell git tag --points-at HEAD)
+
+
+ifneq ("$(which gotestsum)", "")
+	TESTEXE := gotestsum --
+else
+	TESTEXE := go test ./...
+endif
+
+
+BUILD_DATE=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
+ifneq ("$(TAG)", "")
+	VERSION := $(TAG)
+else
+	VERSION := $(COMMIT)-$(BRANCH)-$(BUILD_DATE)
+endif
+
+LDFLAGS = $(shell echo -X main.Version=$(VERSION) -X main.GitCommit=$(COMMIT) -X main.BuildDate=$(BUILD_DATE))
+
+all: test lint build coverage ## test, lint, build, coverage test run
+
+
+.PHONY: build lint test coverage
+lint: ## Run golangci-lint
+	golangci-lint run
+
+coverage: ## Verify the test coverage remains high
+	./scripts/check-coverage.sh 80
+
+test: ## Run tests without coverage
+	rm -rf pkg/grammar/temp
+	$(TESTEXE)
+
+BINARY := sysl
+PLATFORMS := windows linux darwin
+os = $(word 1, $@)
```

Comment:
not sure why you're using `$(word ...)` here. `$@` is the name of the target being built, so it is only one word anyway. I think you should just be able to use `$@` in place of `$(os)` in the recipe below.

Reply:
Not gonna lie copied it straight off some blog post :) I've successful avoided makefiles for just about my whole career :)

Reply:
fixed